commit 1df49008f4ddec9d4f6862b47ea5bdba82078aa4
Author: Akinobu Mita <akinobu.mita@gmail.com>
Date:   Tue Feb 20 13:57:56 2007 -0800

    [PATCH] fault injection: split up stacktrace filter Kconfig option
    
    There is no prompt for CONFIG_STACKTRACE, so FAULT_INJECTION cannot be
    selected without LOCKDEP enabled.  (found by Paolo 'Blaisorblade'
    Giarrusso)
    
    In order to fix such broken Kconfig dependency, this patch splits up the
    stacktrace filter support for fault injection by new Kconfig option, which
    enables to use fault injection on the architecture which doesn't have
    general stacktrace support.
    
    Cc: "Paolo 'Blaisorblade' Giarrusso" <blaisorblade@yahoo.it>
    Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 4448f91b865c..3f3e7403dcac 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -411,8 +411,6 @@ config LKDTM
 config FAULT_INJECTION
 	bool "Fault-injection framework"
 	depends on DEBUG_KERNEL
-	depends on STACKTRACE
-	select FRAME_POINTER
 	help
 	  Provide fault-injection framework.
 	  For more details, see Documentation/fault-injection/.
@@ -440,3 +438,11 @@ config FAULT_INJECTION_DEBUG_FS
 	depends on FAULT_INJECTION && SYSFS && DEBUG_FS
 	help
 	  Enable configuration of fault-injection capabilities via debugfs.
+
+config FAULT_INJECTION_STACKTRACE_FILTER
+	bool "stacktrace filter for fault-injection capabilities"
+	depends on FAULT_INJECTION_DEBUG_FS && STACKTRACE_SUPPORT
+	select STACKTRACE
+	select FRAME_POINTER
+	help
+	  Provide stacktrace filter for fault-injection capabilities